internal/core/useCases/checkout/CheckoutUseCase.go

```diff
@@ -0,0 +1,65 @@
+package checkout
+
+import (
+	"fmt"
+
+	"github.com/hcsouza/fiap-tech-fast-food/internal/core/useCases/order"
+	"github.com/hcsouza/fiap-tech-fast-food/internal/core/valueObject/orderStatus"
+)
+
+type checkoutUseCase struct {
+	order order.IOrderUseCase
+}
+
+func NewCheckoutUseCase(orderUseCase order.IOrderUseCase) ICheckoutUseCase {
+	return &checkoutUseCase{
+		order: orderUseCase,
+	}
+}
+
+func (uc *checkoutUseCase) CreateCheckout(orderId string) (*CreateCheckout, error) {
+	order, err := uc.order.FindById(orderId)
+	nextStatus := orderStatus.ORDER_PAYMENT_PENDING
+
+	if err != nil {
+		return nil, err
+	}
+
+	if !order.OrderStatus.IsValidNextStatus(nextStatus.String()) {
+		return &CreateCheckout{
+			CheckoutURL: "",
+			Message:     "Order already has a checkout completed",
```

Comment:
Será que não faz sentido criar um ERROR pra isso? Porque aqui a message está começando com `Order` em maiusculo e e no `UpdateCheckout` está iniciando com `order` em minúsculo,  e é feito a verificação da string pra este error. Ai acaba ficando em cada lugar de um jeito, e até em testes teria que colocar em cada lugar diferente.

Dentro de `core/errors/errors.go` a gente tem a criação de um error value, ai assim não teria este problema.

`var ErrDuplicatedKey = errors.New("duplicate key error on collection")`

Daria pra usar o `errors.Is(err, ErrDuplicatedKey)`  por exemplo

Reply:
Perfeito Hugo, muito bom!!

Alterei, dá uma olhada se está de acordo, por gentileza 🙏🏻